src/backend/distributed/master/master_repair_shards.c

```diff
@@ -233,6 +233,14 @@ RepairShardPlacement(int64 shardId, char *sourceNodeName, int32 sourceNodePort,
 
 	EnsureTableOwner(distributedTableId);
 
+	/*
+	 * Ensure schema exists on each worker node. We can not run this function
+	 * transactionally, since we may create shards over separate sessions and
+	 * shard creation depends on the schema being present and visible from all
+	 * sessions.
+	 */
+	EnsureSchemaExistsOnAllNodes(distributedTableId);
```

Comment:
There's a small optimisation we could make here to only create the schema on the target node. That would avoid problems when some of the nodes are down.

Reply:
Introduced a new function for this purpose.

`EnsureSchemaExistsOnNode(relationId, nodeName, nodePort)` will now be used to replicate/repair a shard in a single node